lib/receptor_controller/client/response_worker.rb

```diff
@@ -114,6 +118,8 @@ def check_timeouts(threshold = config.response_timeout)
 
         sleep(config.response_timeout_poll_time)
       end
+    rescue => err
+      logger.error("Exception in maintenance worker: #{err}\n#{err.backtrace.join("\n")}")
     end
```

Comment:
Basically I assume the `while started.true` is there in case of this - so it'll enter the loop again endlessly while recovering from exceptions, basically until `started.value == false`.